### PR TITLE
Adds genre to Image, Document, and StudentWork.

### LINKED ISSUE #1102: Research work-type facets vs resource-type facets
In scholar 2.0: We have a facet for work type, with an additional, un-faceted field for genre.

In Scholar 3, we will need a facet for work types, and the Metadata folks are recommending mapping the genre field to resource type, retaining the facet, and expanding its use to other work types (beyond Document and Image).

We need to research and consider how to add work-type facets, and how they balance against resource-type selection and facets.

### LINKED ISSUE #1273: Add genre to Image and Document work types.
### Descriptive summary

Use the resource_type field to map the genre list for Image and Documents in Scholar 2.x to Scholar 3.0.  

Do not introduce genres for other work types yet.  

Remove resource_type from Collections metadata form.

Genre is not faceted.

### PR DESCRIPTION
**Pull request by scherztc**
Fixes #1102 
Fixes #1273 

Summary:

Adds the genre as a controlled vocabulary to Images, Documents, and Student Works as it is in Scholar 2.x.  Removes the resource_type from all work types.

### CODE REVIEW

**Comment by jamesvanmil** [comment]
Two notes - for Genre, we haven't used 'other' as an option.

For Student Works, we have had both a 'Work Type' field and a 'genre' field - did we decide to scrap one of these? I'm only seeing the 'genre' content.

**Comment by scherztc** [comment]
@jamesvanmil :  I removed "Other" for Image and Document, but left it in for Student Work.  Since we got approval from POs to remove complex Student Work faceting, it was my understanding that we would only need the genre field for Student Works.  I kept the genre field and adjusted the label to use Work Types similar to labeling and help in Scholar 2.x.

File: app/helpers/sufia_helper.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def sorted_college_list_for_etds
   def sorted_college_list_for_degrees
     COLLEGE_AND_DEPARTMENT["current_colleges_for_degrees"].keys.collect do |k|
       COLLEGE_AND_DEPARTMENT["current_colleges_for_degrees"][k]["label"]
-    end.sort << "Other"
```
Comment (jamesvanmil) [inline app/helpers/sufia_helper.rb]: We need to keep this "Other"

File: app/helpers/sufia_helper.rb [hunk 2]
```diff
@@ -37,7 +37,7 @@ def sorted_college_list_for_other_works
     )
     list.keys.collect do |k|
       list[k]["label"]
-    end.sort << "Other"
```
Comment (jamesvanmil) [inline app/helpers/sufia_helper.rb]: Need this one "Too"

File: app/views/curation_concerns/base/_attribute_rows.html.erb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 <%= presenter.attribute_to_html(:creator, render_as: :faceted) %>
-<%= presenter.attribute_to_html(:contributor, render_as: :faceted) %>
-<%= presenter.attribute_to_html(:college, render_as: :faceted) %>
-<%= presenter.attribute_to_html(:department, render_as: :faceted) %>
-<%= presenter.attribute_to_html(:admin_set, render_as: :faceted) %>
+<%= presenter.attribute_to_html(:contributor) %>
+<%= presenter.attribute_to_html(:college) %>
```
Comment (jamesvanmil) [inline app/views/curation_concerns/base/_attribute_rows.html.erb]: I think we'd want to keep the faceted label on this and department - keep these as links from the show page?

**Comment by jamesvanmil** [comment]
In Scholar 2, the "Other" label actually mapped to Generic Work on student work types - Other might not make sense in this new context.

Otherwise, I've added some line-wise notes, and the new helper `sorted_genre_list_for_works` needs some spec work.

**Comment by scherztc** [comment]
@jamesvanmil : Ready for review.  I removed Other from Student Work and cleaned up the Help Text, made the line item changes, and wrote some test for the sufia_helper_spec.

**Comment by jamesvanmil** [comment]
Hey sorry for the delay - one other note. The College lists generated by the helper need to hold on to "Other" - this is for folks who aren't in an academic unit.

**Comment by newmanld** [comment]
In regard to genre field, See #1315 and #1314 which asked that we consult with metadata SME.  Did we do so, and did Metadata SME still want a controlled list, or free text with examples in help,  for genre on any or all work types (including ETDs)?  Maintain differences in a controlled list (the list has different choices by work type in Scholar 2.x)?  Is the label meant to be 'type' on one work type (ETD), 'Work type' on another and 'Genre on another'?  Please update metadata spreadsheet with all decisions as well, including values of a drop down list, labels and help text (and examples won't apply if it is changed to a controlled list instead of free text): https://docs.google.com/spreadsheets/d/1biO5sm4WVK9p01jCFDzNqTQ1mF_33MEHbpCjIVaOU3M/edit?usp=sharing.  Thanks.

**Pull request merged**